File: Package.java
public class Package {
    private int id, zipcode, weight;
    private Boolean hazmat;

    // Constructors
    public Package() {
        this.id = 0;
        this.zipcode = 0;
        this.weight = 0;
        this.hazmat = false;
    }
    public Package(int nID, int nZ, int nW, int nH) {
        this.setId(nID);
        this.setZip(nZ);
        this.setWeight(nW);
        this.setHazmat(nH);
    }

    // Setters and Getters
    public void setId(int nID) {
        if (nID != 0)
            this.id = nID;
        else
            this.id = 0;
    }
    public void setZip(int nZ) {
        if (nZ != 0)
            this.zipcode = nZ;
        else
            this.zipcode = 0;
    }
    public void setWeight(int nW) {
        if (nW != 0)
            this.weight = nW;
        else
            this.weight = 0;
    }
    public void setHazmat(int nID) {
        if (nID == 0)
            this.hazmat = false;
        else
            this.hazmat = true;
    }
    public int getID() {
        return this.id;
    }
    public int getZip() {
        return this.zipcode;
    }
    public int getWeight() {
        return this.weight;
    }
    public Boolean getHazmat() {
        return this.hazmat;
    }


    // Member functions
    public String toString() {
        return "Tracking number: " +this.id+" Zipcode: "+this.zipcode+" Weight: "+this.weight+" Hazmat: "+this.hazmat;
    }
}

File: package.cpp
// Copyright 2021 Evan Scales
#include "./package.h"

Package::Package(std::string nID, std::string nZ, int nW, double nV) {
    id = nID;
    zipcode = nZ;
    weight = nW;
    volume = nV;
    tir = 0;
}
void Package::printPackage() const {
    std::cout << "Tracking number: " << id << " Shipping to (zipcode): " << zipcode << " Weight: " << weight << " Volume (cubic feet): " << volume << "\n";
}
std::string Package::getID() const {
    return id;
}
bool Package::getHazmat() const {
    return hazmat;
}
std::string Package::getZip() const {
    return zipcode;
}
int Package::getWeight() const {
    return weight;
}
double Package::getVolume() const {
    return volume;
}
int Package::getTIR() const {
    return tir;
}
void Package::upTIR() {
    tir++;
}
bool Package::operator==(const Package & rhs) const {
    if ((id == rhs.id) && (zipcode == rhs.zipcode) && (weight == rhs.weight) && (volume == rhs.volume))
        return true;
    return false;
}
void Package::resetTIR() {
    tir = 0;
}